apps/blockchain/lib/blockchain/transaction.ex

```diff
@@ -330,11 +331,11 @@ defmodule Blockchain.Transaction do
       iex> Blockchain.Transaction.intrinsic_gas_cost(%Blockchain.Transaction{to: <<>>, init: <<1, 2, 0, 3>>, data: <<>>}, %Block.Header{number: 5_000_000})
       3 * 68 + 4  + 21000
   """
-  @spec intrinsic_gas_cost(t, Header.t()) :: Gas.t()
-  def intrinsic_gas_cost(tx, block_header) do
+  @spec intrinsic_gas_cost(t, EVM.Configuration.t()) :: Gas.t()
+  def intrinsic_gas_cost(tx, config) do
     data_cost = input_data_cost(tx)
-    cc_cost = contract_creation_cost(tx, block_header)
-    data_cost + cc_cost + Gas.g_transaction()
+
+    data_cost + contract_creation_cost(tx, config)
```

Comment:
I see that you have combined the contract creation cost and the cost for the transaction. Should we rename the function so it no longer reads `contract_creation_cost`? As it stands it's a little misleading because this could return the transaction cost and not the contract creation cost.

Reply:
yes, good point